src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.inline.hpp

```diff
@@ -399,7 +399,7 @@ ShenandoahScanRemembered<RememberedSet>::verify_registration(HeapWord* address,
   if (max_offset > CardTable::card_size_in_words) {
     max_offset = CardTable::card_size_in_words;
   }
-  size_t prev_offset;
+  size_t prev_offset = offset;
```

Comment:
Is this change motivated by a detected bug, or is this just a "style" improvement?  It looks to me like this change is equivalent to the original code, because in both cases (when !ctx and ctx), we unconditionally set prev_offset to offset in the first iteration of the do-while loop.

Reply:
Unconditionally setting prev_offset in the second branch (i.e., when `ctx` is not null) _is_ the bug. If the code there sets prev_offset to refer to refer to an _unmarked_ object, the following verification check to get the `size` on an unmarked object could crash. The intention here was to make sure that `prev_offset` never refers to an unmarked object, but this change isn't enough to guarantee that.

Reply:
Thanks for clarifying.  Looks good to me as now drafted.